File: 3903-smallest-stable-index-i/3903-smallest-stable-index-i.java
class Solution {
    public int firstStableIndex(int[] nums, int k) {
        int n = nums.length;
        int min_ind = 0;
        for (int i = 0;i<n;i++){
            int min = Integer.MAX_VALUE;
            int max = Integer.MIN_VALUE;
            for (int j = 0;j<=i;j++){
                max = Math.max(max,nums[j]);
            }
            for (int j = i;j<n;j++){
                min = Math.min(min,nums[j]);
            }
            int stab_score = (max-min);
            if (stab_score<=k){
                return i;
            }
        }
        return -1;
    }
}

File: 3903-smallest-stable-index-i/3903-smallest-stable-index-i.py
class Solution(object):
    def firstStableIndex(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        n = len(nums)
        for i in range(0,n):
            minimum = float('inf')
            maximum = float('-inf')
            for j in range(0,i+1):
                maximum = max(maximum,nums[j])
            for j in  range(i,n):
                minimum = min(minimum,nums[j])
            stability_score = maximum-minimum
            if (stability_score<=k):
                return i
        return -1
                
        